### PR TITLE
Update detailed combat info widget

### LINKED ISSUE #1219: Enhance Win/Loss to combat pop-up
**Is your feature request related to a problem? Please describe.**
When a battle between two nation's units occurs, the client will display something like this:

image

A visually impaired person, such as one that is red/green color blind will not see that the winner of the combat has a green shading in the box and the loser will have a red shading in the box. It is also not clear that the negative number value in the upper right is for hit points.

**Describe the solution you'd like**
In the lower right corner of the box and a white W or L to denote win or loss and then add "HP" in front of the value in the upper right to make what happened more clear.

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by jwrober**
Closes #1219

### CODE REVIEW

File: client/hudwidget.cpp [hunk 1]
```diff
@@ -1792,15 +1794,16 @@ void hud_unit_combat::paintEvent(QPaintEvent *event)
   QFont f = fcFont::instance()->getFont(fonts::default_font);
   QString ahploss, dhploss;
 
+  // TRANS: HP is Hip Point
   if (att_hp_loss > 0) {
-    ahploss = "-" + QString::number(att_hp_loss);
+    ahploss = "-" + QString::number(att_hp_loss) + " HP";
```
Comment (lmoureaux) [inline client/hudwidget.cpp:1799]: It's always better to translate strings in full, in case the order must change (consider currencies, $10 is 10€). The underscore function is the translation:
```suggestion
    ahploss = QString(_("-%1 HP")).arg(att_hp_loss);
```
Same below...

File: client/hudwidget.cpp [hunk 2]
```diff
@@ -1834,17 +1837,24 @@ void hud_unit_combat::paintEvent(QPaintEvent *event)
   p.setFont(f);
   p.setPen(QColor(Qt::white));
   if (def_veteran) {
-    p.drawText(right,
-               Qt::AlignHCenter | Qt::AlignJustify | Qt::AlignAbsolute,
-               QStringLiteral("*"));
+    p.drawText(right, Qt::AlignBottom | Qt::AlignRight | Qt::AlignAbsolute,
+               QStringLiteral("*Vet*"));
```
Comment (lmoureaux) [inline client/hudwidget.cpp:1841]:
```suggestion
               _("*Vet*"));
```

File: client/hudwidget.cpp [hunk 3]
```diff
@@ -1834,17 +1837,24 @@ void hud_unit_combat::paintEvent(QPaintEvent *event)
...
   }
   if (att_veteran) {
-    p.drawText(left, Qt::AlignHCenter | Qt::AlignJustify | Qt::AlignAbsolute,
-               QStringLiteral("*"));
+    p.drawText(left, Qt::AlignBottom | Qt::AlignRight | Qt::AlignAbsolute,
+               QStringLiteral("*Vet*"));
```
Comment (lmoureaux) [inline client/hudwidget.cpp:1845]:
```suggestion
               _("*Vet*"));
```

File: client/hudwidget.cpp [hunk 4]
```diff
@@ -1834,17 +1837,24 @@ void hud_unit_combat::paintEvent(QPaintEvent *event)
...
   p.drawImage(right, dimg);
   p.drawText(right, Qt::AlignHorizontal_Mask, dhploss);
+  if (def_win) {
+    p.drawText(right, Qt::AlignBottom | Qt::AlignLeft | Qt::AlignAbsolute,
+               QStringLiteral("Win"));
+  }
+  if (att_win) {
+    p.drawText(left, Qt::AlignBottom | Qt::AlignLeft | Qt::AlignAbsolute,
+               QStringLiteral("Win"));
+  }
```
Comment (lmoureaux) [inline client/hudwidget.cpp:1857]: Can both `def_win` and `att_win` be true simultaneously? Maybe with `Combat_Rounds` or bombarding?

**Review by lmoureaux — CHANGES_REQUESTED** [review]
Some comments without running the code, mainly around translateability.

File: client/hudwidget.cpp (hunk 4 above)
Comment (jwrober) [reply client/hudwidget.cpp:1857]: I wasn't sure so I went with a variable for both and added them to the `if (defender_hp <= 0)` block

File: client/hudwidget.cpp [hunk 5]
```diff
@@ -1793,14 +1795,15 @@ void hud_unit_combat::paintEvent(QPaintEvent *event)
   QString ahploss, dhploss;
 
   if (att_hp_loss > 0) {
-    ahploss = "-" + QString::number(att_hp_loss);
+    // TRANS: HP - Hit Points
+    ahploss = QString(_("-%1 HP")).arg(att_hp_loss);
   } else {
-    ahploss = QStringLiteral("0");
+    ahploss = QString(_("%1 HP")).arg(att_hp_loss);
```
Comment (lmoureaux) [inline client/hudwidget.cpp:1801]: I had a hard time understanding what the two branches of the `if` do. The `true` branch displays the HP loss if it is positive and the second branch is for no HP loss. Assuming that `att_hp_loss` is never negative, we could simplify this beast to:
```cpp
// TRANS: Unit lost %1 hit points in combat. Keep short.
ahploss = QString(_("-%1 HP")).arg(att_hp_loss);
```
And similar for `dhploss`.

File: client/hudwidget.cpp [hunk 6]
```diff
@@ -1834,17 +1837,24 @@ void hud_unit_combat::paintEvent(QPaintEvent *event)
   p.setFont(f);
   p.setPen(QColor(Qt::white));
   if (def_veteran) {
-    p.drawText(right,
-               Qt::AlignHCenter | Qt::AlignJustify | Qt::AlignAbsolute,
-               QStringLiteral("*"));
+    p.drawText(right, Qt::AlignBottom | Qt::AlignRight | Qt::AlignAbsolute,
+               _("*Vet*"));
```
Comment (lmoureaux) [inline client/hudwidget.cpp:1841]: What are the stars for? Maybe change the color to something goldy and leave a hint for translators?

File: client/hudwidget.cpp [hunk 7]
```diff
@@ -1834,17 +1837,24 @@ void hud_unit_combat::paintEvent(QPaintEvent *event)
...
-               QStringLiteral("*"));
+    p.drawText(left, Qt::AlignBottom | Qt::AlignRight | Qt::AlignAbsolute,
+               _("*Vet*"));
   }
   p.drawText(left, Qt::AlignHorizontal_Mask, ahploss);
   p.drawImage(right, dimg);
   p.drawText(right, Qt::AlignHorizontal_Mask, dhploss);
+  if (def_win) {
+    p.drawText(right, Qt::AlignBottom | Qt::AlignLeft | Qt::AlignAbsolute,
+               _("Win"));
```
Comment (lmoureaux) [inline client/hudwidget.cpp:1852]: I suggest `Winner` here and/or leave a note for translators (`Win`/`Gagne` would make no sense, we really want `Victor`/`Vainqueur` in French).

**Review by lmoureaux — COMMENTED** [review]
With the basic things out of the way it was easier to review the logic. I'm convinced that the code works. Some things could be improved or simplified here and there but I won't oppose merging as is.

File: client/hudwidget.cpp (hunk 7 above)
Comment (lmoureaux) [reply client/hudwidget.cpp:1852]: Or maybe `Wins`...

File: client/hudwidget.cpp (hunk 5 above)
Comment (jwrober) [reply client/hudwidget.cpp:1801]: I mostly just kept what was there and only modified to suit the issue. I wasn't looking to improve/refactor anything. I can look at this, however.

File: client/hudwidget.cpp (hunk 6 above)
Comment (jwrober) [reply client/hudwidget.cpp:1841]: As you can see, the original `drawText()` simply showed a star for a vet, which wasn't clear. I can look at an alternate color or simply change to remove the stars.

File: client/hudwidget.cpp (hunk 7 above)
Comment (jwrober) [reply client/hudwidget.cpp:1852]: I'll change to "Winner" and add a //TRANS

File: client/hudwidget.cpp (hunk 5 above)
Comment (jwrober) [reply client/hudwidget.cpp:1801]: I think we need both sides. 
```  
  if (att_hp_loss > 0) {
    // TRANS: HP - Hit Points
    ahploss = QString(_("-%1 HP")).arg(att_hp_loss);
  } else {
    ahploss = QString(_("%1 HP")).arg(att_hp_loss);
  }
```
If HP loss is greater than 0 (meaning the unit took a hit and lost some HP), then we show a negative number in the text. We then assume in the `else` statement that there was no loss to HP and give a "positive" number that will essentially say "0 HP". I think this is more about the text setup. We don't want to show "-0 HP".

Comment (lmoureaux) [reply client/hudwidget.cpp:1801]: I kind of assumed that `-0 HP` was fine (would be useful to distinguish lost HP from remaining HP). Otherwise it can be `ahploss = QString(_("%1 HP")).arg(-att_hp_loss)`

**Review by lmoureaux — APPROVED** [review]
